tools/integration/test/integration/e2e-test-service/originsTest.js

```diff
@@ -0,0 +1,113 @@
+const assert = require('assert')
+const { callFetch } = require('../../../lib/fetch')
+const { devApiBaseUrl, prodApiBaseUrl, getComponents, definition } = require('../testConfig')
+
+const ORIGIN_EXCLUSION_LIST = ['go/golang', 'debsrc/debian', 'maven/mavengoogle']
+const ORIGIN_REVISIONS_EXCLUSION_LIST = ['debsrc/debian']
+
+const MAVEN_COMPONENT_GROUP_ID = 'maven/mavencentral/org.apache.httpcomponents'
+const MAVEN_COMPONENT_PARTIAL_GROUP_ID = 'maven/mavencentral/org.apache.httpcompon'
+const GRADLE_COMPONENT_ENDPOINT = 'gradleplugin/io.github.lognet.grpc-spring-boot'
+
+;(async function validateOriginsApi() {
+  const components = await getComponents()
+
+  describe('Validate Origins API Between Dev and Prod', function () {
+    this.timeout(definition.timeout)
+
+    afterEach(() => new Promise(resolve => setTimeout(resolve, definition.timeout / 2)))
+
+    components.filter(isOriginAllowed).forEach(component => {
+      it(`Validates Origins API response for ${component}`, () => compareOrigins(component))
+    })
+
+    components.filter(isOriginWithRevisionsAllowed).forEach(component => {
+      it(`Validates Origins API response with revisions for ${component}`, () => compareOriginsWithRevisions(component))
+    })
+
+    it('Validates Origins API response for a Maven component with only a group ID', () =>
+      compareOrigins(MAVEN_COMPONENT_GROUP_ID))
+
+    it('Validates Origins API response for a Maven component with a partial group ID for suggestion checks', () =>
+      compareOrigins(MAVEN_COMPONENT_PARTIAL_GROUP_ID))
+
+    it('Validates Origins API response for a Gradle plugin component', () =>
+      compareEndpoints(GRADLE_COMPONENT_ENDPOINT))
+
+    it('Validates Origins API with revisions response for a Gradle plugin component', () =>
+      compareEndpoints(`${GRADLE_COMPONENT_ENDPOINT}/revisions`))
+  })
+})()
+
+function extractIds(response) {
+  return response.map(item => item.id)
+}
+
+function assertResponsesMatch(actual, expected) {
+  const sortedActualIds = extractIds(actual).sort()
+  const sortedExpectedIds = extractIds(expected).sort()
+
+  assert.deepStrictEqual(sortedActualIds, sortedExpectedIds)
+}
+
+function isCoordinateAllowed(coordinate, exclusionList) {
+  return !exclusionList.some(excluded => coordinate.includes(excluded))
+}
+
+function isOriginAllowed(coordinate) {
+  return isCoordinateAllowed(coordinate, ORIGIN_EXCLUSION_LIST)
+}
+
+function isOriginWithRevisionsAllowed(coordinate) {
+  return isCoordinateAllowed(coordinate, ORIGIN_REVISIONS_EXCLUSION_LIST)
+}
+
+function getProviderType(type, provider) {
+  switch (type) {
+    case 'git':
+      return 'github'
+    case 'gem':
+      return 'rubygems'
+    case 'conda':
+      return `conda/${provider}`
+    case 'maven':
+      return provider === 'mavengoogle' ? provider : type
```

Comment:
what about maven for mavencentral, mavengoogle and gradleplugin is the same as provider?

Reply:
```
case 'git':
case 'gem':
    return provider
```
What do you think?